### PR TITLE
Fix bugs

### PR DESCRIPTION
**Pull request by torkleyy**
Turns out I actually used this pattern 3 times..

### CODE REVIEW

File: src/de/tests.rs [hunk 1]
```diff
@@ -159,3 +159,16 @@ fn untagged() {
 
     assert_eq!(from_str::<Untagged>("true").unwrap(), Untagged::Bool(true));
 }
+
+#[test]
+fn forgot_apostrophes() {
+    let de: Result<(i32, String)> = from_str("(4, \"Hello)");
+
+    assert!({
+        if let Err(Error::Parser(ParseError::ExpectedStringEnd, _)) = de {
```
Comment (kvark) [inline src/de/tests.rs]: match ftw!

Comment (torkleyy) [reply src/de/tests.rs]: I wondered why this isn't easier, but how does match help here?

Comment (Xaeroxe) [reply src/de/tests.rs]: (He might be referencing the concept of pattern matching?)

Comment (kvark) [reply src/de/tests.rs]: I just don't see the point in `if let` sugar over `match` if you take both arms and occupy more lines in the end

**Comment by Xaeroxe** [comment]
Ping @kvark

It'd be really nice to get this big fix released as a patch version.

**Comment by torkleyy** [comment]
@Xaeroxe that's my fault, didn't address the review earlier.

**Pull request merged**